### PR TITLE
enhance(view): context menu for tree view

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
- This PR aims to add a context menu for tree view. Added Delete Command in tree view context menu.
- It also fixes an issue with Delete Command: If the note is not open in the text editor. it is not deleted when the command is run from the explorer menu.
- It adds a Create Note command in context menu of Tree view, the lookup bar has the initial value of tree item responsible for action.
- adds source: Tree view property for Create Note and Delete Command

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [ ] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [x] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [~] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [x] [Write Tests](dev.process.qa.test) 
- [ ] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [x] Common cases tested
- [x] 1-2 Edge cases tested
- [~] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

## Docs
- [~] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/plugin-core/src/commands/DeleteCommand.ts [hunk 1]
```diff
@@ -73,6 +68,60 @@ export class DeleteCommand extends BasicCommand<CommandOpts, CommandOutput> {
     return nodePosition?.end.line;
   }
 
+  /**
+   * When Delete Command is ran from Tree view, it gets Noteprops as args
+   */
+  private isNotePropsArgs(opts: CommandOpts) {
```
Comment (hkievet) [inline packages/plugin-core/src/commands/DeleteCommand.ts:74]: suggestion:  Maybe make a commandUtils.ts file or something under commands that could share this function between the two commands you have here?

**Review by hkievet — COMMENTED** [review]
Good job getting this to work!  I'm glad you were able to teach me more about this part of the code base when we paired together!

File: packages/plugin-core/src/commands/CreateNoteCommand.ts [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
+import {
+  CommandOutput as NoteLookupOutput,
+  CommandRunOpts as NoteLookupRunOpts,
+} from "./NoteLookupCommand";
 import { AutoCompletableRegistrar } from "../utils/registers/AutoCompletableRegistrar";
+import _ from "lodash";
+import { window } from "vscode";
 
-type CommandOpts = {};
+type CommandOpts = any;
```
Comment (kevinslin) [inline packages/plugin-core/src/commands/CreateNoteCommand.ts:12]: can we apply the proper types to this?

File: packages/plugin-core/src/test/suite-integ/NativeTreeView.test.ts [hunk 3]
```diff
@@ -117,6 +117,7 @@ suite("NativeTreeView tests", function () {
             genRandomId: true,
           });
         },
+        timeout: 1e6,
```
Comment (kevinslin) [inline packages/plugin-core/src/test/suite-integ/NativeTreeView.test.ts:120]: is this necessary?
